easybuild/easyblocks/generic/modulerc.py

```diff
@@ -0,0 +1,94 @@
+##
+# Copyright 2018-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing a software-specific .modulerc file
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.config import build_option
+from easybuild.tools.filetools import write_file
+
+
+class ModuleRC(EasyBlock):
+    """
+    Generic easyblock to create a software-specific .modulerc file
+    """
+
+    def configure_step(self):
+        """Do nothing."""
+        pass
+
+    def build_step(self):
+        """Do nothing."""
+        pass
+
+    def install_step(self):
+        """Do nothing."""
+        pass
+
+    def make_module_step(self, fake=False):
+        """Install .modulerc file."""
+        modfile_path = self.module_generator.get_module_filepath(fake=fake)
+        modulerc = os.path.join(os.path.dirname(modfile_path), '.modulerc')
+
+        if os.path.exists(modulerc) and not build_option('force'):
```

Comment:
This implies that you can only have one entry in the `.modulerc` per software package. Probably ok, but worth noting the limitation

Reply:
Indeed, that's a rather strict limitation for now, which we can review later when the need for it pops up?

A whole bunch of questions come into play to relax this:

* What do we do with an existing `.modulerc` that was not created by EasyBuild?
* How do we allow both appending to an existing `.modulerc` and overwriting the existing one (from the `eb` command line, I don't think specifying this in the easyconfig makes much sense)?
* Should we detect potentially conflicting lines in an existing `.modulerc`?

Reply:
Indeed, that's a rather strict limitation for now, which we can review later when the need for it pops up?

A whole bunch of questions come into play to relax this:

* What do we do with an existing `.modulerc` that was not created by EasyBuild?
* How do we allow both appending to an existing `.modulerc` and overwriting the existing one (from the `eb` command line, I don't think specifying this in the easyconfig makes much sense)?
* Should we detect potentially conflicting lines in an existing `.modulerc`?